airflow/api_connexion/security.py

```diff
@@ -14,24 +14,27 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-
 from functools import wraps
 from typing import Callable, Optional, Sequence, Tuple, TypeVar, cast
 
 from flask import Response, current_app
 
 from airflow.api_connexion.exceptions import PermissionDenied, Unauthenticated
+from airflow.api_connexion.webserver_auth import requires_authentication
 
 T = TypeVar("T", bound=Callable)  # pylint: disable=invalid-name
 
 
 def check_authentication() -> None:
     """Checks that the request has valid authorization information."""
     response = current_app.api_auth.requires_authentication(Response)()
-    if response.status_code != 200:
+    jwt_response = requires_authentication(Response)()
```

Comment:
This is not going to be accepted like this, sorry.

It means you _have_ to log in to API before you can use it which represents a massive breaking change in the behaviour to the API from 2.0.0/.1.

Reply:
I'm trying to figure out the best way to plug the jwt into the app. What I did here was to try both auth_backend and jwt_response. I have updated the failing path which made the auth_backend from 2.0.0/.1 unusable. Now both the old auth and the new one works. The major problem now is that all authentication failures now appear as 401. I'm still thinking around it because it'll be good to tell the user that the token expired, invalid token or malformed authorization code.

If you can check again to see that both now works. I also added a comment on why error is not raised for expired token